packages/batch/tests/unit/BatchProcessor.test.ts

```diff
@@ -0,0 +1,477 @@
+/**
+ * Test BatchProcessor class
+ *
+ * @group unit/batch/class/batchprocessor
+ */
+import { BatchProcessingError, BatchProcessor, EventType } from '../../src';
+import { DynamoDBRecord, KinesisStreamRecord, SQSRecord } from 'aws-lambda';
+import {
+  sqsEventFactory,
+  kinesisEventFactory,
+  dynamodbEventFactory,
+} from '../../tests/helpers/factories';
+import {
+  sqsRecordHandler,
+  asyncSqsRecordHandler,
+  kinesisRecordHandler,
+  asyncKinesisRecordHandler,
+  dynamodbRecordHandler,
+  asyncDynamodbRecordHandler,
+} from '../../tests/helpers/handlers';
+
+describe('Class: BatchProcessor', () => {
+  const ENVIRONMENT_VARIABLES = process.env;
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+    jest.resetModules();
+    process.env = { ...ENVIRONMENT_VARIABLES };
+  });
+
+  afterAll(() => {
+    process.env = ENVIRONMENT_VARIABLES;
+  });
+
+  describe('Synchronously processing SQS Records', () => {
+    test('Batch processing SQS records with no failures', async () => {
+      // Prepare
+      const firstRecord = sqsEventFactory('success');
+      const secondRecord = sqsEventFactory('success');
+      const records = [firstRecord, secondRecord];
+      const processor = new BatchProcessor(EventType.SQS);
+
+      // Act
+      processor.register(records, sqsRecordHandler);
+      const processedMessages = await processor.process();
+
+      // Assess
+      expect(processedMessages).toStrictEqual([
+        ['success', (firstRecord as SQSRecord).body, firstRecord],
```

Comment:
Since the return type of the factory is already `SQSRecord` perhaps we don't need a typecast here.